tests/Utf8Json.Tests/DoubleConversionTest.cs

```diff
@@ -115,5 +124,22 @@ public void Float()
                 GetString(item).Contains("E").IsTrue();
             }
         }
+
+        [Theory]
```

Comment:
Addign more test cases like this:

```
        public static IEnumerable<object[]> CheckStringFormatProviderData()
        {
            foreach (string cultureName in new string[] { "en-US", "de-DE" })
            {
                foreach (string dStr in new string[] { "59.08634249999999", "-59.08634249999999", "1.7976931348623157E+308", "-1.7976931348623157E+308", "NaN", "Infinity", "-Infinity" })
                {
                    yield return new object[] { cultureName, dStr };
                }
            }
        }

        [Theory]
        [MemberData(nameof(CheckStringFormatProviderData))]
        public void CheckStringFormatProvider(string cultureName, string dStr)
        {
            // cause this number has more digits than kMaxExactDoubleIntegerDecimalDigits ComputeGuess will fail and the fallback will process this float-point value
            CurrentCulture = CreateSpecificCulture(cultureName);

            var d = double.Parse(dStr, InvariantCulture);
            var buf = Encoding.UTF8.GetBytes(dStr);
            var d2 = NumberConverter.ReadDouble(buf, 0, out var _);

            Approximately(d, d2).IsTrue();
        }
```

However test cases Infinity, -Infinity and NaN fail. Is that expected?